src/generic/Matrix.jl

```diff
@@ -5167,3 +5173,32 @@ function change_base_ring(M::MatrixElem, R::AbstractAlgebra.Ring)
    end
    return N
 end
+
+@doc Markdown.doc"""
+    map!(f, dst::Generic.MatrixElem, src::Generic.MatrixElem)
+
+> Like `map`, but stores the result in `dst` rather than a new matrix.
+"""
+function Base.map!(f, dst::MatrixElem, src::MatrixElem)
```

Comment:
Is everyone else happy with f being the first, rather than the last argument to map? I think this is different to the convention we have elsewhere, e.g. change_base_ring, map_from_func, etc. I guess we are following a Julia convention here? @thofma @fieker

Reply:
> I guess we are following a Julia convention here?

Yes, generally function arguments come first: https://docs.julialang.org/en/v1/manual/style-guide/index.html#Write-functions-with-argument-ordering-similar-to-Julia-Base-1
Moreover, specifically for `map`, which is a `Base` function, it it would be too annoying to have different methods with diverging convention.

Reply:
I was unaware of this convention. We should probably change the functions we have that violate it. I'll open a ticket for that, although it will be a pain to change because it is by now used in Hecke.jl and Singular.jl too.